src/game/overlay.c

```diff
@@ -214,98 +278,114 @@ void Overlay_Init()
     for (int i = 0; i < MAX_PICKUPS; i++) {
         m_Pickups[i].duration = 0;
     }
+
+    Overlay_SetupHealthBar();
+    Overlay_SetupAirBar();
+    Overlay_SetupEnemyBar();
+}
+
+void Overlay_SetHealtBarTimer(int16_t health_bar_timer)
+{
+    m_HealthBar.timer = health_bar_timer;
+}
+
+void Overlay_TimerTick()
+{
+    m_HealthBar.timer--;
+    if (m_HealthBar.timer < 0) {
+        Overlay_SetHealtBarTimer(0);
+    }
 }
 
 void Overlay_DrawHealthBar()
 {
     static int32_t old_hit_points = 0;
 
-    for (int i = 0; i < 6; i++) {
+    for (int32_t i = 0; i < 6; i++) {
```

Comment:
Let's leave it as `int` IMO, although no strong feeling about this

Reply:
Ok. I changed it since most of the ints are given defined types and it seemed inconsistent. But it shouldn't matter for a loop counter.